src/app/components/MediaLoader/utils/buildSettings.ts

```diff
@@ -0,0 +1,60 @@
+import getPlayerProps from '#app/legacy/containers/MediaPlayer/helpers/propsInference';
+import onClient from '#app/lib/utilities/onClient';
+import { BuildConfigProps } from '../types';
+
+const isTestURL = () => {
+  let isTestRender = false;
+
+  if (onClient()) {
+    const queryParams = new URLSearchParams(window.location.search);
+    isTestRender = queryParams.get('renderer_env') === 'test';
+  }
+
+  return isTestRender;
+};
```

Comment:
Just thinking, do we need to use the Test mediator if we're on the Test environment in general? We tend to not use the `renderer_env` flag on Test, as the environment variables should determine what to use behind the scenes.

Reply:
I put this one in to apply this setting: `{ mediator: { host: 'open.test.bbc.co.uk' } }`

Reply:
I figured that way, we could easily switch between live and test assets when doing development on our localhost.

Reply:
I could switch it the other way round so that `{ mediator: { host: 'open.test.bbc.co.uk' } }` is applied by default, but removed if it's on live?

Reply:
I think its ok for localhost, I'm just thinking when its deployed to Test, will we need to remember to add in `renderer_env=test` to get the video player to work correctly?

Reply:
My gut feeling is that the mediator value should default to the environment its on and then we override it with the `renderer_env` query param.

So on local/test it defaults to `{ mediator: { host: 'open.test.bbc.co.uk' } }` and we use `renderer_env` flags to override it if needed.
On Live `mediator` is removed entirely and cannot be overridden.

Reply:
Coolio

Reply:
Something along these lines may work:
```
switch (true) {
  // Conditions to use Test Mediator
  case queryParams.get('renderer_env') === 'test':
  case isLocal():
  case isTest():
    return true;
  default:
    return false;
}
```